pkg/skel/skel.go

```diff
@@ -180,11 +181,21 @@ func (t *dispatcher) pluginMain(cmdAdd, cmdDel func(_ *CmdArgs) error, versionIn
 	}
 
 	if err != nil {
+		configVersion, verr := t.ConfVersionDecoder.Decode(cmdArgs.StdinData)
+		if verr != nil {
+			configVersion = "0.2.0"
+		}
 		if e, ok := err.(*types.Error); ok {
 			// don't wrap Error in Error
+			if e.CniVersion == "" {
+				e.CniVersion = configVersion
+			}
 			return e
 		}
-		return createTypedError(err.Error())
+
+		e := createTypedError(err.Error())
```

Comment:
Could we just have createTypedError take a version string which it then sets in the error?